File: app/src/main/java/com/evanzheng/bibliobarcode/AuthorTextWatcher.java
package com.evanzheng.bibliobarcode;

import android.text.Editable;
import android.text.TextWatcher;

public abstract class AuthorTextWatcher implements TextWatcher {

    int id;

    AuthorTextWatcher(int id) {
        this.id = id;
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
    }


    @Override
    public void afterTextChanged(Editable s) {
    }
}

File: app/src/main/java/com/evanzheng/bibliobarcode/AuthorTextWatcher.kt
package com.evanzheng.bibliobarcode

import android.text.Editable
import android.text.TextWatcher

//A custom implementation of an TextWatcher with an id field.
internal abstract class AuthorTextWatcher(val id: Int) : TextWatcher {
    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
    override fun afterTextChanged(s: Editable) {}
}